broadinstitute/single_cell_portal_core
# Move @fullhuman/postcss-purgecss to devDependencies (SCP-5223)

This remediates a reported [security issue](https://defectdojo.dsp-appsec.broadinstitute.org/finding/97980), brought to our attention by @sarahgibs.

DSP's security scanning tool DefectDojo detected `inflight 1.0.6` as insecure.  That's in SCP due to `@fullhuman/postcss-purgecss 4.1.3 > purgecss 4.1.3 > glob 7.2.3 > inflight 1.0.6`.  It turns out that that [@fullhuman/postcss-purgecss](https://www.npmjs.com/package/@fullhuman/postcss-purgecss) JS package was declared in `package.json` under `dependencies`, but can just as well be declared in `devDependencies`.  So it's just a development dependency, and not needed in production.

To test:
* Pull branch
* Freshly install via `yarn install --force`
* Restart Rails and Vite
* Run JS tests via `yarn test`
* Click around in app, confirm no adverse effects.  [PurgeCSS](https://purgecss.com/) removes unused CSS, so beyond simply completely failing to load any page, I'd expect any effects here to be visual defects.  I saw no issues in my local testing.

This satisfies SCP-5223.
